aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -567,7 +569,9 @@ Tensor &nuclear_norm_out(Tensor& result, const Tensor& self, bool keepdim) {
       "Expected a tensor with 2 dimensions, but got a ",
       self.dim(),
       " dimensions tensor instead.");
-  return at::sum_out(result, std::get<1>(at::svd(self)), 0, keepdim);
+  bool self_need_grad = self.is_variable() && self.requires_grad();
+  auto S = std::get<1>(at::svd(self, /*some=*/true, /*compute_uv=*/self_need_grad));
```

Comment:
I really don’t get it. Why do you have to specify the value of this argument in the forward function? Is running this function with compute_uv false an error if the input requires grad or what?

Reply:
`compute_uv` is required to be `true` if one wants to backprop through `svd` (yes it's an error otherwise). So if the input of forward don't require grad, we can skip it because we only use `s`.